File: ui/kp-textile/src/app/models/transporter-model.ts
import { AddressDto, ContactDetailsDto } from "./client-model";
import { CommonListResponse, CommonResponse } from "./genric-model";

export interface TransporterDto {
    branchId: string;
    sortKey: string;
    transporterId: string;
    companyName: string;
    alias: string;
    contactInfo: ContactDetailsDto;
    website: string;
    paymentTerm: string;
    addresses: AddressDto[];
    remark: string;
    status: string;
    gstn: string;
    parentTransporterId: string;
    parentTransporterPath: string;
}

export interface TransporterContactPersonDto {
    branchId: string;
    sortKey: string;
    transporterId: string;
    contactId: string;
    salutation: string;
    firstName: string;
    lastName: string;
    contactInfo: ContactDetailsDto;
    department: string;
    personType: string;
    remark: string;
    address: AddressDto;
}

export interface AddTransporterRequest extends TransporterDto {
    contactPersons: TransporterContactPersonDto[]
}

export interface AddTransporterResponse extends CommonResponse {
    data: AddTransporterRequest
}

export interface TransporterFilterDto {
    branchId: string;
    companyName: string;
    alias: string;
    email: string;
    contactPersonFirstName: string;
    contactPersonLastName: string;
    paymentTerm: string;
}

export interface TransporterListRequest extends TransporterFilterDto {
    lastEvalutionKey: any
    pageSize: number

}

export interface TransporterListResponse extends CommonListResponse {
    data: TransporterDto[];
}

export interface TransporterRequest {
    id: string;
}

export interface TransporterResponse extends CommonResponse {
    data: TransporterDto;
}

export interface GetTransporterRequestDto {
    branchId: string;
    transporterId: string;
}

File: commonpkg/models/transporter-model.go
package models

import (
	"github.com/aws/aws-sdk-go/service/dynamodb"
)

type TransporterDto struct {
	TransporterId         string            `json:"transporterId,omitempty" uri:"transporterId"`
	SortKey               string            `json:"sortKey,omitempty"`
	CompanyName           string            `json:"companyName,omitempty"`
	Alias                 string            `json:"alias,omitempty"`
	ContactInfo           ContactDetailsDto `json:"contactInfo,omitempty"`
	Website               string            `json:"website,omitempty"`
	PaymentTerm           string            `json:"paymentTerm,omitempty"`
	Addresses             []AddressDto      `json:"addresses,omitempty"`
	Remark                string            `json:"remark,omitempty"`
	Status                string            `json:"status,omitempty"`
	Gstn                  string            `json:"gstn,omitempty"`
	ParentTransporterId   string            `json:"parentTransporterId,omitempty"`
	ParentTransporterPath string            `json:"parentTransporterPath,omitempty"`
}

type TransporterContactPersonDto struct {
	TransporterId string            `json:"transporterId,omitempty"`
	SortKey       string            `json:"sortKey,omitempty"`
	ContactId     string            `json:"contactId,omitempty"`
	Salutation    string            `json:"salutation,omitempty"`
	FirstName     string            `json:"firstName,omitempty"`
	LastName      string            `json:"lastName,omitempty"`
	ContactInfo   ContactDetailsDto `json:"contactInfo,omitempty"`
	Department    string            `json:"department,omitempty"`
	PersonType    string            `json:"personType,omitempty"`
	Remark        string            `json:"remark,omitempty"`
	Address       AddressDto        `json:"address,omitempty"`
}

type AddTransporterRequest struct {
	TransporterDto
	ContactPersons []TransporterContactPersonDto `json:"contactPersons,omitempty"`
}

type AddTransporterResponse struct {
	CommonResponse
	Data AddTransporterRequest `json:"data,omitempty"`
}

type TransporterFilterDto struct {
	CompanyName            string `json:"companyName,omitempty" form:"companyName"`
	Alias                  string `json:"alias,omitempty" form:"alias"`
	Email                  string `json:"email,omitempty" form:"email"`
	ContactPersonFirstName string `json:"contactPersonFirstName,omitempty" form:"contactPersonFirstName"`
	ContactPersonLastName  string `json:"contactPersonLastName,omitempty" form:"contactPersonLastName"`
	PaymentTerm            string `json:"paymentTerm,omitempty" form:"paymentTerm"`
}

type TransporterListRequest struct {
	LastEvalutionKey map[string]*dynamodb.AttributeValue `json:"lastEvalutionKey,omitempty"`
	PageSize         int64                               `json:"pageSize,omitempty" form:"pageSize"`
	TransporterFilterDto
}

type TransporterListResponse struct {
	CommonListResponse
	Data []TransporterDto `json:"data,omitempty"`
}

type TransporterRequest struct {
	Id string `uri:"id"`
}

type TransporterResponse struct {
	CommonResponse
	Data TransporterDto `json:"data,omitempty"`
}

type GetTransporterRequestDto struct {
	TransporterId string `uri:"transporterId"`
}
